avro4s-core/src/test/scala/com/sksamuel/avro4s/github/Github415.scala

```diff
@@ -3,25 +3,28 @@ package com.sksamuel.avro4s.github
 import java.io.{FileOutputStream, ObjectOutputStream}
 
 import com.sksamuel.avro4s.Encoder
+import com.sksamuel.avro4s.github.Github415.PlaybackSession
 import org.scalatest.funsuite.AnyFunSuite
 import org.scalatest.matchers.must.Matchers
 
 class Github415 extends AnyFunSuite with Matchers {
 
   test("github 415") {
-
-    object Rebuffers {
-      case class Metrics(count: Int)
-      case class EarlyLate(early: Metrics)
-      case class Stats(session: Option[EarlyLate])
-    }
-
-    case class Rebuffers(network: Option[Rebuffers.Stats])
-
-    case class PlaybackSession(rebuffers: Option[Rebuffers])
-
+    // TODO clarify is it necessary to be able to serialize encoders / decoders of unserializable classes?
```

Comment:
the Rebuffers classes aren't serializable as it they are inside a non-serializable class (Github415). For the test I moved them therefore to the companion object. Why this was relevant is something I don't remember fully, but iirc this was due to the codecs referring to the Magnolia case classes that referenced the class.